remote-keypair/src/openpgp_card.rs

```diff
@@ -0,0 +1,359 @@
+                    {
+                        return Err(SignerError::Connection(format!(
+                            "expected Ed25519 key, got {:?}",
+                            ecc_attrs.curve()
+                        )));
+                    }
+                    pk.data().try_into().map_err(|e| {
+                        SignerError::Connection(format!("key on card is malformed: {e}"))
+                    })?
+                }
+                _ => {
+                    return Err(SignerError::Connection(
+                        "expected ECC key, got RSA".to_string(),
+                    ))
+                }
+            },
+            _ => {
+                return Err(SignerError::Connection(
+                    "expected ECC key, got RSA".to_string(),
+                ))
+            }
+        };
+
+        Ok(Pubkey::from(pk_bytes))
+    }
+
+    fn try_sign_message(&self, message: &[u8]) -> Result<Signature, SignerError> {
+        // Verify that card has valid signing key
+        self.try_pubkey()?;
+
+        let mut pgp_mut = self.pgp.borrow_mut();
+        let opt = &mut pgp_mut.transaction().map_err(|e| {
+            SignerError::Connection(format!("could not start transaction with card: {e}"))
+        })?;
+        let card_info: OpenpgpCardInfo = opt
+            .try_into()
+            .map_err(|e| SignerError::Connection(format!("could not get card info: {e}")))?;
+
+        // Prompt user for PIN verification if and only if
+        //   * Card indicates PIN is only valid for one PSO:CDS command at a time, or
+        //   * PIN has not yet been entered for the first time.
+        if card_info.pin_cds_valid_once || !*self.pin_verified.borrow() {
+            let mut pin = get_pin_from_user_as_bytes(&card_info, true)?;
+            while opt.verify_pw1_sign(pin.as_bytes()).is_err() {
+                pin = get_pin_from_user_as_bytes(&card_info, false)?;
+            }
+            *self.pin_verified.borrow_mut() = true;
+        }
+
+        // Await user touch confirmation if and only if
+        //   * Card supports touch confirmation, and
+        //   * Touch policy set anything other than "off".
+        if let Some(signing_uif) = card_info.signing_uif {
+            if signing_uif.touch_policy().touch_required() {
+                println!("Awaiting touch confirmation...");
+            }
+        }
+
+        // Delegate message signing to card
+        let hash = Hash::EdDSA(message);
+        let sig = opt
+            .signature_for_hash(hash)
+            .map_err(|e| SignerError::Protocol(format!("card failed to sign message: {e}")))?;
+
+        Ok(Signature::new(&sig[..]))
+    }
+
+    fn is_interactive(&self) -> bool {
+        true
+    }
+}
+
+/// Data struct for convenience.
+#[derive(Debug)]
+struct OpenpgpCardInfo {
+    aid: ApplicationIdentifier,
+    cardholder_name: String,
+    signing_uif: Option<UIF>,
+    pin_cds_valid_once: bool,
+}
```

Comment:
I think this doc-comment does not really bring any clarity.
Consider removing it or expanding with some details.
For example, what exactly this "convenience" is about.